packages/compiler-sfc/src/script/resolveType.ts

```diff
@@ -39,9 +39,10 @@ import { parse as babelParse } from '@babel/parser'
 import { parse } from '../parse'
 import { createCache } from '../cache'
 import type TS from 'typescript'
-import { dirname, extname, join } from 'path'
+import { dirname, extname, isAbsolute, join } from 'path'
 import { minimatch as isMatch } from 'minimatch'
 import * as process from 'process'
+import { imports as resolveImports } from 'resolve.exports'
```

Comment:
Is it necessary to use `resolve.exports` here? We have already imported `minimatch` here, so perhaps we can manually match the file paths. Reference: https://github.com/vuejs/core/blob/a28794edfabe403f4c595f48d150894c5c6c1ed4/packages/compiler-sfc/src/script/resolveType.ts#L1042

Reply:
Ah, I suppose that's possible but subpath imports have a few additional complexities such as supporting [conditional export syntax](https://nodejs.org/api/packages.html#conditional-exports) and [nested conditions](https://nodejs.org/api/packages.html#nested-conditions), along with some other minor syntax differences like needing to start with `#`, etc. I'm happy to implement these if adding an additional dependency is preferred to be avoided but I figured this library effectively handles all of this already.